google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/AbstractSqlScriptVerifier.java

```diff
@@ -0,0 +1,452 @@
+
+  private final Map<String, Object> variables = new HashMap<>();
+
+  /**
+   * Constructor for a verifier that will take a {@link GenericConnection} as a parameter to the
+   * {@link AbstractSqlScriptVerifier#verifyStatementsInFile(GenericConnection, String, Class, boolean)}
+   */
+  public AbstractSqlScriptVerifier() {
+    this(null);
+  }
+
+  /** Constructor for a verifier that will use a connection provider for connections */
+  public AbstractSqlScriptVerifier(GenericConnectionProvider provider) {
+    this.connectionProvider = provider;
+  }
+
+  /**
+   * Reads sql statements from the specified file name and executes and verifies these. Statements
+   * that are preceeded by an @EXPECT statement are verified against the @EXPECT specification.
+   * Statements without an @EXPECT statement will be executed and its result will be ignored, unless
+   * the statement throws an exception, which will fail the test case.
+   *
+   * The {@link Connection}s that the statements are executed on must be created by a
+   * {@link GenericConnectionProvider}
+   *
+   * @param filename The file name containing the statements. Statements must be separated by a
+   *        semicolon (;)
+   * @param resourceClass The class that should be used to locate the resource specified by the file
+   *        name
+   * @param logStatements Should the verifier log each statement that is executed and verified to
+   *        standard out
+   * @throws Exception
+   */
+  public void verifyStatementsInFile(String filename, Class<?> resourceClass, boolean logStatements)
+      throws Exception {
+    verifyStatementsInFile(connectionProvider.getConnection(), filename, resourceClass,
+        logStatements);
+  }
+
+  /**
+   * Reads sql statements from the specified file name and executes and verifies these. Statements
+   * that are preceeded by an @EXPECT statement are verified against the @EXPECT specification.
+   * Statements without an @EXPECT statement will be executed and its result will be ignored, unless
+   * the statement throws an exception, which will fail the test case.
+   *
+   * @param connection The {@link Connection} to execute the statements against
+   * @param filename The file name containing the statements. Statements must be separated by a
+   *        semicolon (;)
+   * @param resourceClass The class that defines the package where to find the input file
+   * @param logStatements Should all statements be logged to standard out?
+   */
+  public void verifyStatementsInFile(GenericConnection connection, String filename,
+      Class<?> resourceClass, boolean logStatements) throws Exception {
+    try {
+      List<String> statements = readStatementsFromFile(filename, resourceClass);
+      for (String statement : statements) {
+        String sql = statement.trim();
+        if (logStatements) {
+          System.out.println("\n------------------------------------------------------\n"
```

Comment:
should we switch this to logging?

Reply:
No, I think it's better to keep it like this to avoid depending on any logging framework. This is only used to generate new test SQL scripts if something (fundamental) changes in the JDBC driver, such as:
* Adding new statements to the client side SQL dialect.
* Changing the behavior of existing client side SQL statements.